### PR TITLE
Couple minor optimisations in Kernel

### PR DESCRIPTION
**Pull request by michalmuskala**

### CODE REVIEW

**Comment by josevalim** [comment]
@michalmuskala please take a look at the build error.

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -4573,7 +4576,7 @@ defmodule Kernel do
   end
 
   defp split_words(string, [mod])
-  when mod == ?s or mod == ?a or mod == ?c do
+       when mod === ?s or mod === ?a or mod === ?c do
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex]: Let's revert this one. The style inconsistency with the codebase is not worth the very small gains.

**Comment by michalmuskala** [comment]
Updated

**Comment by josevalim** [comment]
@michalmuskala please check the build. :)

**Comment by michalmuskala** [comment]
@josevalim with the failure, I think moving to the map syntax will produce a significantly inferior message, so we should skip it and keep `Map.replace!` as it is right now.

**Comment by josevalim** [comment]
@michalmuskala sounds good to me!

**Comment by michalmuskala** [comment]
I reverted to use `Map.replace!` where it makes a difference in the error message.

I also changed what's the `__struct__/1` implementation in case there's no `@enforce_keys` set. We don't need to track it.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**